src/backend/query_builder.rs

```diff
@@ -1378,29 +1344,58 @@ pub trait QueryBuilder: QuotedBuilder + EscapeBuilder + TableRefBuilder {
         right: &SimpleExpr,
         sql: &mut dyn SqlWriter,
     ) {
-        let no_paren = matches!(op, BinOper::Equal | BinOper::NotEqual);
-        let left_paren = left.need_parentheses()
-            && left.is_binary()
-            && op != left.get_bin_oper().unwrap()
-            && !no_paren;
+        // If left has higher precedence than op, we can drop parentheses around left
+        let drop_left_higher_precedence =
+            inner_greater_precedence_than_outer(left, &(*op).into()).unwrap_or(false);
+
+        let known_assoc_op = matches!(
+            op,
+            BinOper::And | BinOper::Or | BinOper::Add | BinOper::Sub | BinOper::Mul | BinOper::Mod
+        );
+
+        #[cfg(feature = "backend-postgres")]
+        let assoc_op = known_assoc_op || matches!(op, BinOper::PgOperator(PgBinOper::Concatenate));
+
+        #[cfg(not(feature = "backend-postgres"))]
+        let assoc_op = known_assoc_op;
```

Comment:
I think we can create a new trait method, have a default impl in this file, which is then overrided by the specific Postgres backend. Sort of like how `quote()` works.

Reply:
Will fix!

Reply:
Fixed now! Learned something new :-)